relentless/core.py

```diff
@@ -99,6 +106,50 @@ def __call__(self, x):
 
         return result
 
+    def derivative(self, x, n):
+        """Evaluate the n-th derivative of the interpolating function.
+
+        Parameters
+        ----------
+        x : float or array_like
+            1-d array of x coordinates to evaluate.
+        n : int
+            The order of the derivative to take.
+
+        Returns
+        -------
+        result : float or numpy.ndarray
+            Interpolated derivative values having the same form as `x`.
+
+        Raises
+        ------
+        ValueError
+            If n is not a positive integer.
+
+        """
+        if not isinstance(n, int) and n <= 0:
+            raise ValueError('n must be a positive integer')
+        scalar_x = np.isscalar(x)
+        x = np.atleast_1d(x)
+        result = np.zeros(len(x))
+
+        # clamp lo
+        lo = x < self.domain[0]
+        result[lo] = self._spline(self.domain[0], nu=n)
+
+        # clamp hi
+        hi = x > self.domain[1]
+        result[hi] = self._spline(self.domain[1], nu=n)
```

Comment:
We define the extrapolation to be a constant outside the domain, so the derivative should be zero for `x < self.domain[0]` and `x > self.domain[1]`.